### PR TITLE
More better names

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

**Review by RAOF — COMMENTED** [review]
I have a weak preference for `DisplaySink` over `DisplayDevice`, as my feeling is “Sink” is more aligned with the domain terminology, but `DisplayDevice` is definitely an improvement.

**Comment by AlanGriffiths** [comment]
> I have a weak preference for `DisplaySink` over `DisplayDevice`

I started out with `DisplaySink` but, as I looked at the surrounding code, I changed my mind.

@Saviq @mattkae - time to pile in with additional opinions

File: include/platform/mir/graphics/platform.h [hunk 1]
```diff
@@ -163,8 +163,8 @@ class GLRenderingProvider : public RenderingProvider
 
     virtual auto as_texture(std::shared_ptr<Buffer> buffer) -> std::shared_ptr<gl::Texture> = 0;
 
-    virtual auto surface_for_output(
```
Comment (RAOF) [inline include/platform/mir/graphics/platform.h:166]: Does this need to be renamed?

If it does, maybe it should be harmonised with `GLRenderingProvider::suitability_for_display`?

Comment (AlanGriffiths) [reply include/platform/mir/graphics/platform.h:166]: Well, maybe I read it wrong, but it reads better to me:

```c++
    auto output_surface = chosen_allocator->surface_for_device(
        display_device, display_device.pixel_size(), *gl_config);
```

File: src/platforms/gbm-kms/server/buffer_allocator.cpp [hunk 2]
```diff
@@ -540,7 +540,7 @@ auto mgg::GLRenderingProvider::surface_for_output(
         size);
 }
 
-auto mgg::GLRenderingProvider::make_framebuffer_provider(DisplayBuffer& /*target*/)
+auto mgg::GLRenderingProvider::make_framebuffer_provider(DisplayDevice& /*target*/)
```
Comment (mattkae) [inline src/platforms/gbm-kms/server/buffer_allocator.cpp:543]: Should this be `/*device*/`?

File: src/platforms/renderer-generic-egl/buffer_allocator.cpp [hunk 3]
```diff
@@ -425,17 +425,17 @@ auto mge::GLRenderingProvider::suitability_for_display(
     return probe::unsupported;
 }
 
-auto mge::GLRenderingProvider::surface_for_output(
-    DisplayBuffer& target,
+auto mge::GLRenderingProvider::surface_for_device(
+    DisplayDevice& target,
```
Comment (mattkae) [inline src/platforms/renderer-generic-egl/buffer_allocator.cpp:429]: Should this be `device`?

File: src/platforms/renderer-generic-egl/buffer_allocator.cpp [hunk 4]
```diff
@@ -444,7 +444,7 @@ auto mge::GLRenderingProvider::surface_for_output(
         size);
 }
 
-auto mge::GLRenderingProvider::make_framebuffer_provider(DisplayBuffer& /*target*/)
+auto mge::GLRenderingProvider::make_framebuffer_provider(DisplayDevice& /*target*/)
```
Comment (mattkae) [inline src/platforms/renderer-generic-egl/buffer_allocator.cpp:447]: Should this be `device`?

**Review by mattkae — COMMENTED** [review]
Some nits, but it is what I expected

**Review by Saviq — APPROVED** [review]
I'm not going to help a lot, but I have no strong preference between `DisplaySink` and `DisplayDevice`.

My only concern with Device would be the potential for ambiguity. But that can be improved by means of documentation.

File: include/platform/mir/graphics/platform.h (hunk 1 above)
Comment (AlanGriffiths) [reply include/platform/mir/graphics/platform.h:166]: Or, maybe this suggests another route? `DisplayOutput`?

**Comment by AlanGriffiths** [comment]
> My only concern with Device would be the potential for ambiguity. But that can be improved by means of documentation.

If `DisplaySink` is less prone to ambiguity, I think that's the better approach. (I just found it overly generic)

**Comment by Saviq** [comment]
> If `DisplaySink` is less prone to ambiguity, I think that's the better approach. (I just found it overly generic)

I do think it is, I can think of multiple _devices_ (DRM, input, hardware…) that we already deal with. But if `Sink` isn't right, either - maybe we need to come up with something else again. We _do_ sink renderables into it, though, so :shrug:.

File: src/server/compositor/multi_threaded_compositor.cpp [hunk 5]
```diff
@@ -98,10 +98,10 @@ class CompositingFunctor
         //Appease TSan, avoid destructor and this thread accessing the same shared_ptr instance
         auto const disp_listener = display_listener;
         auto display_registration = mir::raii::paired_calls(
-            [this, &disp_listener]{group.for_each_display_buffer([&disp_listener](mg::DisplayBuffer& buffer)
-                { disp_listener->add_display(buffer.view_area()); });},
-            [this, &disp_listener]{group.for_each_display_buffer([&disp_listener](mg::DisplayBuffer& buffer)
-                { disp_listener->remove_display(buffer.view_area()); });});
+            [this, &disp_listener]{ group.for_each_display_sink([&disp_listener](mg::DisplaySink& sink)
```
Comment (Saviq) [inline src/server/compositor/multi_threaded_compositor.cpp:101]:
```suggestion
            [this, &disp_listener]{group.for_each_display_sink([&disp_listener](mg::DisplaySink& sink)
```

File: tests/unit-tests/platforms/gbm-kms/kms/CMakeLists.txt [hunk 6]
```diff
@@ -18,7 +18,7 @@ mir_add_wrapped_executable(mir_unit_tests_gbm-kms NOINSTALL
 set_property(
   SOURCE test_platform.cpp test_graphics_platform.cpp test_buffer_allocator.cpp
          test_display.cpp test_display_generic.cpp test_display_multi_monitor.cpp test_display_configuration.cpp
-         test_display_buffer.cpp test_drm_helper.cpp
+        test_display_sink.cpp test_drm_helper.cpp
```
Comment (Saviq) [inline tests/unit-tests/platforms/gbm-kms/kms/CMakeLists.txt:21]:
```suggestion
         test_display_sink.cpp test_drm_helper.cpp
```

**Review by mattkae — APPROVED** [review]
I cannot think of anything better than sink, so I am happy

**Review by Saviq — APPROVED** [review]
Just a couple whitespace nits from me.

Feel free to merge when done.

**Pull request merged**